JamfSync/Model/Jcds2Dp.swift

```diff
@@ -200,33 +207,453 @@ class Jcds2Dp: DistributionPoint {
         } else {
             fileUrl = file.fileUrl
         }
+        
         guard let fileUrl else { throw DistributionPointError.badFileUrl }
+        
+        
+        _ = disableSleep(reason: "start upload")
+
+        if let fileProperties = try? FileManager.default.attributesOfItem(atPath: fileUrl.path(percentEncoded: false)) {
+            if let size = fileProperties[FileAttributeKey.size] as? NSNumber {
+                let uploadFileSize = size.doubleValue
+                Chunk.all = Int(truncating: size)
+                Chunk.numberOf = Int(uploadFileSize)/Chunk.size
+                if Chunk.all % Chunk.size > 0 {
+                    Chunk.numberOf += 1
+                }
+                LogManager.shared.logMessage(message: "File will be split into \(Chunk.numberOf) parts.", level: .debug)
+                if uploadFileSize > 32212255000 {
+                    LogManager.shared.logMessage(message: "Maximum upload file size (30GB) exceeded. File size: \(Int(uploadFileSize))", level: .info)
+                    return
+                }
+            }
+        } else {
+            LogManager.shared.logMessage(message: "A problem occurred trying to access the file: \(fileUrl)", level: .debug)
```

Comment:
Best to be descriptive about what kind of problem occurred. Instead of doing `if let fileProperties = try? ...` add a 
```
do {
    try ...
} catch {
    LogManager.shared.logMessage(message: "A problem occurred trying to access the file \"\(fileUrl)\": \(error)", level: .error)
}
```
Also note that the level should be .error since it's probably going to cause the upload to fail. Unless it's not, in which case maybe just .warning.